code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -2228,7 +2228,29 @@
 	to_chat(exposed_human, span_notice("Your hair starts growing at an incredible speed!"))
 	exposed_human.hairstyle = "Very Long Hair"
 	exposed_human.facial_hairstyle = "Beard (Very Long)"
-	exposed_human.update_hair()
+	exposed_human.update_hair(is_creating = TRUE)
+
+/datum/reagent/concentrated_barbers_aid/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	. = ..()
+	if(current_cycle > 20 / creation_purity)
+		if(!ishuman(M))
+			return
+		var/mob/living/carbon/human/human_mob = M
+		if(creation_purity == 1 && human_mob.has_quirk(/datum/quirk/item_quirk/bald))
+			human_mob.remove_quirk(/datum/quirk/item_quirk/bald)
+		var/datum/species/species_datum = human_mob.dna?.species
+		if(!istype(species_datum))
+			return
+		if(species_datum.species_traits.Find(HAIR))
```

Comment:
```suggestion
		if(hair in species_datum.species_traits))
```

Reply:
Do you mean `HAIR`?

Reply:
I thought the `list.Find(thing)` syntax was faster than the `thing in list` syntax.

Reply:
Not sure but this code is really cold so it doesn't matter anyway

Choose whatever you think looks best, though we use `in` most of the time